File: frontend/packages/types/src/partner-offer-readiness.ts
export type PartnerOfferValueCategory =
  | "percent_discount"
  | "free_item"
  | "loyalty_reward"
  | "exclusive_access"
  | "event_benefit"
  | "welcome_bonus";

export type PartnerOfferReadinessLevel = "ready" | "partial" | "not_ready";

export type PartnerOfferReadinessCheckSeverity = "ok" | "warning" | "error";

export interface PartnerOfferReadinessCheck {
  key: string;
  label: string;
  passed: boolean;
  severity: PartnerOfferReadinessCheckSeverity;
}

export interface PartnerOfferReadinessFields {
  readiness: PartnerOfferReadinessLevel;
  is_passport_eligible: boolean;
  is_placeholder: boolean;
  value_category: PartnerOfferValueCategory;
  value_category_label: string;
  human_description: string;
  checks: PartnerOfferReadinessCheck[];
}

File: backend/app/schemas/partner_offer_readiness.py
"""Partner offer readiness API schemas — RF-02A."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

PartnerOfferReadinessLevel = Literal["ready", "partial", "not_ready"]
PartnerOfferReadinessCheckSeverity = Literal["ok", "warning", "error"]


class PartnerOfferReadinessCheckItem(BaseModel):
    model_config = ConfigDict(extra="forbid")

    key: str
    label: str
    passed: bool
    severity: PartnerOfferReadinessCheckSeverity


class PartnerOfferReadinessFields(BaseModel):
    model_config = ConfigDict(extra="forbid")

    readiness: PartnerOfferReadinessLevel
    is_passport_eligible: bool
    is_placeholder: bool
    value_category: str
    value_category_label: str
    human_description: str
    checks: list[PartnerOfferReadinessCheckItem] = Field(default_factory=list)
